scipy/spatial/transform/rotation.py

```diff
@@ -8,7 +8,106 @@
 AXIS_TO_IND = {'x': 0, 'y': 1, 'z': 2}
 
 
-def _make_elementary_quat(axis, angles):
+def elementary_basis_vector(axis):
+    b = np.zeros(3)
+    b[AXIS_TO_IND[axis]] = 1
+    return b
+
+
+def compute_euler_from_dcm(dcm, seq, extrinsic=False):
+    # The algorithm assumes intrinsic frame transformations. For representation
+    # the paper uses transformation matrices, which are transpose of the
+    # direction cosine matrices used by our Rotation class.
+    # Adapt the algorithm for our case by
+    # 1. Transposing our dcm representation
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+
+    if extrinsic:
+        seq = seq[::-1]
+
+    if dcm.ndim == 2:
+        dcm = dcm[None, :, :]
+    num_rotations = dcm.shape[0]
+
+    # Step 0
+    # Algorithms dcms are transpose of ours
+    dcm = np.transpose(dcm, (0, 2, 1))
+    angles = np.empty((num_rotations, 3))
+    # Algorithm assumes axes as column vectors, here we use 1D vectors
+    n1 = elementary_basis_vector(seq[0])
+    n2 = elementary_basis_vector(seq[1])
+    n3 = elementary_basis_vector(seq[2])
+
+    # Step 2
+    sl = np.dot(np.cross(n1, n2), n3)
+    cl = np.dot(n1, n3)
+
+    # angle offset is lambda from the paper referenced in [2] from docstring of
+    # `as_euler` function
+    offset = np.arctan2(sl, cl)
+    c = np.empty((3, 3))
+    c[0] = n2
+    c[1] = np.cross(n1, n2)
+    c[2] = n1
+
+    # Step 3
+    rt = np.array([
+        [1, 0, 0],
+        [0, cl, -sl],
+        [0, sl, cl]
+    ])
+    rtc = rt.dot(c)
+    ct = c.T
+    res = np.einsum('...ij,...jk->...ik', rtc, dcm)
+    o = np.einsum('...ij,...jk->...ik', res, ct)
+
+    # Step 4
+    phi = np.arccos(o[:, 2, 2])
```

Comment:
I suggested to assign angles column right here, without introducing phi. Then update it as `angles[:, 1] += offset`, when we can. I think now that it is probably worse, decide yourself.

Reply:
Oh, I understand now. Making the change.